netplugin/agent/agent.go

```diff
@@ -60,8 +60,10 @@ func NewAgent(pluginConfig *plugin.Config) *Agent {
 
 	// Initialize appropriate plugin
 	switch opts.PluginMode {
+	case "docker-compose":
```

Comment:
Its really not based on version. Docker versions >1.13/17.03 can have both workflows. In fact, customers deploying Nomad may want to continue using the older workflow.
In this new mode, we avoid creating docker networks implicitly from contiv. By setting this mode users know they are following the new workflow.